RedHat-UX/red-hat-design-system
# refactor(footer): use screen size controller

## What I did

1. Updated footer to use `ScreenSizeController`

What this doesn't do, is use updated tokens imported from the token repo.  That will be in a later PR.

Closes #513

## Testing Instructions

1. View [deploy preview](https://deploy-preview-807--red-hat-design-system.netlify.app/elements/footer/demo/) to ensure mobile state works to swap between accordion and lists views
